### DISCUSSION TITLE
Is it possible to use Conext Providers/ "local" theme?

### DISCUSSION THREAD

**Discussion by mkraina** [jpudysz/react-native-unistyles#308] [Q&A]
As per https://reactnativeunistyles.vercel.app/reference/theming/ 
> With Unistyles 2.0, we no longer use a React Context. However, this doesn’t mean you can’t change the theme. In fact, it’s even easier as you don’t need to re-render the entire app. To change the theme at any time, simply call setTheme function

If I understand correctly, this changes the theme at global level and for whole app, but in some scenarios I'd like to change the theme for certain components and all the children, which won't be possible with the approach above.

### The use case
Let's say I have Dialog that should always be light. That is doable by simply setting the backgroundColor, but as the default text color would be taken from the global palette, I would end up in all underlying Text components having whitish color because the global theme is dark.

Inspecting the `useStyles` / `useUnistyles` code it shouldn't be that hard to add support for this

**Comment by jpudysz** [comment]
I'm no longer focusing on 2.x. Unistyles 3.0 is right around the corner. There will be no hooks, so I can't scope it to anything. This request is on my to-do list, but there's been no progress or ETA so far.
